File: src/main/java/com/dongfg/project/api/component/quartz/builder/CronJob.java
package com.dongfg.project.api.component.quartz.builder;

import lombok.Builder;
import lombok.Data;
import org.quartz.Job;
import org.quartz.JobDataMap;
import org.quartz.TriggerListener;
import org.quartz.listeners.JobListenerSupport;
import org.quartz.listeners.SchedulerListenerSupport;

import java.util.Date;

/**
 * @author dongfg
 * @date 17-12-21
 */
@Data
@Builder
public class CronJob {
    private String group;
    private String name;

    private Class<? extends Job> jobClass;
    private JobDataMap jobDataMap;

    private String cronExpression;
    private Date startTime;
    private Date endTime;

    private JobListenerSupport jobListener;
    private SchedulerListenerSupport schedulerListener;
    private TriggerListener triggerListener;
}

File: src/main/kotlin/com/dongfg/project/api/component/quartz/CronJob.kt
package com.dongfg.project.api.component.quartz

import org.quartz.Job
import org.quartz.JobDataMap
import org.quartz.TriggerListener
import org.quartz.listeners.JobListenerSupport
import org.quartz.listeners.SchedulerListenerSupport
import java.util.*


/**
 * @author dongfg
 * @date 2018/3/16
 */
data class CronJob(
        var name: String? = null,

        var jobClass: Class<out Job>,
        var jobDataMap: JobDataMap? = JobDataMap(),

        var cronExpression: String,
        var startTime: Date? = null,
        var endTime: Date? = null,

        var jobListener: JobListenerSupport? = null,
        var schedulerListener: SchedulerListenerSupport? = null,
        var triggerListener: TriggerListener? = null
)